mongodbatlas/search.go

```diff
@@ -31,6 +31,7 @@ type SearchService interface {
 	ListIndexes(ctx context.Context, groupID string, clusterName string, databaseName string, collectionName string, opts *ListOptions) ([]*SearchIndex, *Response, error)
 	GetIndex(ctx context.Context, groupID, clusterName, indexID string) (*SearchIndex, *Response, error)
 	CreateIndex(ctx context.Context, projectID, clusterName string, r *SearchIndex) (*SearchIndex, *Response, error)
+	CreateIndexWithJSON(ctx context.Context, projectID, clusterName string, body interface{}) (*SearchIndex, *Response, error)
```

Comment:
naming is hard 🤦

Reply:
For context. the variable body is anything that can be converted to JSON, for instance `map[string]interface{}` or even `*SearchIndex`

Reply:
[q] Is it dangerous to just take raw json in here, rather than flags? I'm thinking of the potential to inject or send in other undesired data.

Also doesn't this make validation of the request harder? JSON comes in and all we end up saying is "it's failed, sorry"